erts/emulator/test/float_SUITE.erl

```diff
@@ -56,6 +56,13 @@ otp_7178(Config) when is_list(Config) ->
     {'EXIT', {badarg,_}} = (catch list_to_float("1.0e83291083210")),
     ok.
 
+negative_zero(Config) when is_list(Config) ->
+    <<16#8000000000000000:64>> = <<(my_apply(erlang, '-', [0.0]))/float>>,
+    <<16#8000000000000000:64>> = <<(my_apply(erlang, '*', [-1, 0.0]))/float>>,
+    <<16#8000000000000000:64>> = <<(my_apply(erlang, '*', [-1.0, 0.0]))/float>>,
+    <<16#8000000000000000:64>> = <<(my_apply(erlang, '*', [-1.0, 0]))/float>>,
```

Comment:
I'll just add that I wrote a C test case and ran it on three different architectures (x86-64, SPARC64, and PowerPC64), and they all agree that negating a floating-point zero sets the sign bit. I haven't read chapter-and-verse from the Erlang spec, but I think people expect compatible behaviour from Erlang.